Use Cases/migrate-normal-table-to-timescale-table.md

```diff
@@ -0,0 +1,144 @@
+    "account_id" uuid NOT NULL,
+    "symbol" text,
+    "income_type" text NOT NULL,
+    "income" numeric,
+    "asset" text,
+    "info" text,
+    "time" timestamptz,
+    "time_unix" int8 NOT NULL,
+    "trade_id" text,
+    "tran_id" text,
+    "is_locked_position" bool DEFAULT false,
+    "is_notify" bool DEFAULT false,
+    "created_at" timestamptz NOT NULL DEFAULT now(),
+    "updated_at" timestamptz NOT NULL DEFAULT now(),
+    CONSTRAINT "ts_future_incomes_account_id_fkey" FOREIGN KEY ("account_id") REFERENCES "public"."accounts"("id"),
+    PRIMARY KEY ("id","account_id","income_type","time_unix")
+);
+
+-- Indices
+CREATE UNIQUE INDEX ts_future_incomes_account_id_income_type_tran_id_time_unix_inde ON public.ts_future_incomes USING btree (account_id, income_type, tran_id, time_unix);
+CREATE INDEX ts_future_incomes_account_id_income_type_time_index ON public.ts_future_incomes USING btree (account_id, income_type, "time");
+CREATE INDEX ts_future_incomes_account_id_symbol_income_type_time_index ON public.ts_future_incomes USING btree (account_id, symbol, income_type, "time");
+
+SELECT create_hypertable(
+  'ts_future_incomes',
+  'time_unix',
+  chunk_time_interval => 604800000, -- 7 days in milliseconds
+  create_default_indexes => false
+)
+```
+
+## Create migrator to import data from old table from new timescale table
+
+```mermaid
+stateDiagram-v2
+    [*] --> Initialize: start_link
+    Initialize --> CheckState: init
+
+    state CheckState {
+        [*] --> CountRecords
+        CountRecords --> SetEndTime
+        SetEndTime --> PrepareState
+    }
+
+    CheckState --> MigrationProcess: handle_continue
+
+    state MigrationProcess {
+        [*] --> FetchBatch: Query UserTrades
+        FetchBatch --> TransformData
+        TransformData --> UpdateEndTime: Check max_batch_time
+        UpdateEndTime --> InsertData: Insert to TsUserTrades
+
+        state InsertData {
+            [*] --> NoNewRecords: count = 0
+            [*] --> HasNewRecords: count > 0
+            NoNewRecords --> Complete
+            HasNewRecords --> UpdateOffset
+        }
+    }
+
+    MigrationProcess --> MigrationProcess: Continue next batch
+    MigrationProcess --> [*]: Migration complete
+
+    state State {
+        total_user_trades
+        total_ts_user_trades
+        offset
+        end_time
+    }
+```
+
+## Replace old by new step by step
+
+### Duplicate insert function same as old table
+To make sure new tables have new data same as old tables we insert both of tables to make sure we don't lost new data and can rollback to old table if we have problem
+
+### Clone all query function from old table when replace all current code are using old query
+We need to replace new query to old query function by function to retest to make sure correct data and acceptable query time
+
+### Remove old tables to reduce storage size
+After everything is work fine, we can consider to remove old tables if needed to save data storage
```

Comment:
map into migration plan.
- dual write
- backfilling
- validate
- change primary tables to new -> remove old tables.

Should place before migrator. Migrator is a part of migration plan ?